java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java

```diff
@@ -349,7 +349,9 @@ public void click() {
     }
 
     public void submit() {
+      dispatcher.beforeClickOn(element, driver);
```

Comment:
this shouldn't be beforeClickOn, but rather beforeSubmit? and added to WebDriverEventListener.

Since submit does not synthesize the 'click' events, this isn't accurate.